### PR TITLE
Optional transformer function in TreeSitterClient

### PR DESCRIPTION
**Pull request by kaunteya**
Inclusions

- TreeSitterClient can be initialized only with a language parameter. [Non-breaking change]

```swift
treeSitterClient = try! TreeSitterClient(language: language)
```

----

- `locationTransformer` function will be an optional function.[ Breaking change]

```swift
public let locationTransformer: Point.LocationTransformer?
```

### CODE REVIEW

File: Sources/TreeSitterClient/TreeSitter+Extensions.swift [hunk 1]
```diff
@@ -15,12 +15,8 @@ extension InputEdit {
...
         }
 
-        guard
-            let startPoint = transformer(startLocation),
-            let newEndPoint = transformer(newEndLocation)
-        else {
-            return nil
-        }
+        let startPoint = transformer?(startLocation) ?? .zero
+        let newEndPoint = transformer?(newEndLocation) ?? .zero
```
Comment (mattmassicotte) [inline Sources/TreeSitterClient/TreeSitter+Extensions.swift:19]: This is a subtle change. I'd like to make a clear distinction between the `transformer` being nil, and it being non-nil and returning a nil value.

File: Sources/TreeSitterClient/TreeSitterClient.swift [hunk 2]
```diff
@@ -117,11 +117,7 @@ extension TreeSitterClient {
                                  limit: Int,
                                  readHandler: @escaping Parser.ReadBlock,
                                  completionHandler: @escaping () -> Void) {
-        guard let oldEndPoint = oldEndPoint else {
-            assertionFailure("oldEndPoint unavailable")
-            return
-        }
```
Comment (mattmassicotte) [inline Sources/TreeSitterClient/TreeSitterClient.swift:123]: This assertion should probably remain if locationTransformer is non-nil.

File: Sources/TreeSitterClient/TreeSitterClient.swift [hunk 3]
```diff
@@ -117,11 +117,7 @@ extension TreeSitterClient {
...
-
+        let oldEndPoint = oldEndPoint ?? .zero
         self.oldEndPoint = nil
```
Comment (mattmassicotte) [inline Sources/TreeSitterClient/TreeSitterClient.swift:121]: There's now some slightly confusing variable shadowing going on here. Any ideas on how to improve that?

**Review by mattmassicotte — CHANGES_REQUESTED** [review]

File: Sources/TreeSitterClient/TreeSitterClient.swift (hunk 2 above)
Comment (kaunteya) [reply Sources/TreeSitterClient/TreeSitterClient.swift:123]: Updated code for this change

File: Sources/TreeSitterClient/TreeSitterClient.swift (hunk 3 above)
Comment (kaunteya) [reply Sources/TreeSitterClient/TreeSitterClient.swift:125]: Added self on the property to make it more clear

File: Sources/TreeSitterClient/TreeSitter+Extensions.swift (hunk 1 above)
Comment (kaunteya) [reply Sources/TreeSitterClient/TreeSitter+Extensions.swift]: Update the code to accommodate the mentioned change

**Comment by kaunteya** [comment]
Matt. I have updated the README as well. Please have a look

File: Sources/TreeSitterClient/TreeSitter+Extensions.swift [hunk 4]
```diff
@@ -15,18 +15,25 @@ extension InputEdit {
...
+        assert(startPoint != nil, "startPoint should not be nil")
+        assert(newEndPoint != nil, "newEndPoint should not be nil")
         self.init(startByte: UInt32(range.location * 2),
                   oldEndByte: UInt32(range.max * 2),
                   newEndByte: UInt32(newEndLocation * 2),
-                  startPoint: startPoint,
+                  startPoint: startPoint!,
                   oldEndPoint: oldEndPoint,
-                  newEndPoint: newEndPoint)
+                  newEndPoint: newEndPoint!)
```
Comment (mattmassicotte) [inline Sources/TreeSitterClient/TreeSitter+Extensions.swift:37]: I think we can pull this off with a little less code. What do you think about this?

```swift
let startPoint = transformer?(startLocation)
let endPoint = transformer?(newEndLocation)

if transformer != nil {
    assert(startPoint != nil)
    assert(newEndPoint != nil)
}
```

and then in the initializer, we can use use optional chaining:

```
startPoint: startPoint ?? .zero
```

**Review by mattmassicotte — CHANGES_REQUESTED** [review]

File: Sources/TreeSitterClient/TreeSitter+Extensions.swift (hunk 4 above)
Comment (kaunteya) [reply Sources/TreeSitterClient/TreeSitter+Extensions.swift]: Less code without losing brevity. Looks perfect.
I have updated the code

**Pull request merged**